nmpolicy/internal/resolver/path.go

```diff
@@ -32,12 +32,13 @@ type mapEntryVisitFn func(map[string]interface{}, string) (interface{}, error)
 
 type pathVisitor struct {
 	path              []ast.Node
+	currentStep       ast.Node
```

Comment:
This can be an `int` with the `path` index so we don't have to change `v pathVisitor` to `v* pathVisitor` only because the struct has too much stuff, and alternative is to make `currentStep` a pointer.

Reply:
changed `currentStep` to be a pointer.